src/components/common/IconButton.jsx

```diff
@@ -0,0 +1,57 @@
+import React from 'react';
+import { useState } from 'react';
+import styled from 'styled-components';
+import check from '@/assets/check.svg';
+import plus from '@/assets/plus.svg';
+
+const Styled = {
+  BackGround: styled.div`
+    display: flex;
+    padding: 1.6rem;
+    border-radius: 100px;
+    background: ${({ status, theme }) => {
+      switch (status) {
+        case 'Enabled':
+          return theme.color.btnGr;
+        case 'Disabled':
+          return theme.color.mainGr;
+        case 'Hover':
+          return '#4A4A4A';
+        case 'Pressed':
+          return '#3A3A3A';
+        case 'Focus':
+          return '#3A3A3A';
+        default:
+          return theme.color.btnGr;
+      }
+    }};
+    border: ${({ status }) => (status === 'Focus' ? '1px solid #2B2B2B' : '')};
+  `,
+};
+
+function IconButton({ status, shape }) {
+  const [buttonStatus, setButtonStatus] = useState('Enabled');
+  const handleMouseEnter = () => setButtonStatus('Hover');
+  const handleMouseLeave = () => setButtonStatus('Enabled');
+  const handleMouseDown = () => setButtonStatus('Pressed');
+  const handleFocus = () => setButtonStatus('Focus');
+  const handleBlur = () => setButtonStatus('Enabled');
```

Comment:
해당 함수들은 이벤트가 일어 났을 때 상태에 따라 변경되고 있는 것 같습니다.
모두 setButtonStatus()를 사용하고 있는 것 같아서 
```suggestion
const handlButtonStateChange = (newStatus) => {
  setButtonStatus(newStatus);
}  
```
이런식으로 중복을 줄여보는 것도 좋을 것 같아요!